File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    //variables and user input
    float letters = 0;
    float words = 0;
    float sentences = 0;
    float gradeLevel = 0;

    string sampleText = get_string("Input text sample: ");

    //count letters, words, and sentences
    for (int i = 0, n = strlen(sampleText); i < n; i++)
    {
        //get number of letters
        if ((sampleText[i] >= 'a' && sampleText[i] <= 'z') || (sampleText[i] >= 'A' && sampleText[i] <= 'Z'))
        {
            letters++;
        }

        //get number of words
        if (sampleText[i] == ' ')
        {
            words++;
        }

        if ((sampleText[i] == '.') || (sampleText[i] == '!') || (sampleText[i] == '?'))
        {
            sentences++;
        }
    }
    if (letters > 0)
    {
        words++;
    }

    //Calculations
    gradeLevel = round((0.0588 * ((letters * 100) / words)) - (0.296 * ((sentences * 100) / words)) - 15.8);

    //TEST PRINT printf("%f\n%f\n%f\n", letters, words, sentences);

    //print results
    if (gradeLevel < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (gradeLevel >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", gradeLevel);
    }
}

File: readability.py
from cs50 import get_string
import math


def readingLevel():
    # input
    sampleText = get_string("Input text sample: ").lower()
    # count letters, words and sentences
    letters, words, sentences = (0.0, 0.0, 0.0)
    
    for i in sampleText:
        # letters
        if i.isalpha() is True:
            letters += 1
        # words
        if i == " ":
            words += 1
        # sentences
        if i == '.' or i == '!' or i == '?':
            sentences += 1
    if letters > 0:
        words += 1
    # calculate reading level
    gradeLevel = round((0.0588 * ((letters * 100) / words)) - (0.296 * ((sentences * 100) / words)) - 15.8)
    # print results
    if gradeLevel < 1:
        print("Before Grade 1")
    elif gradeLevel >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {gradeLevel}")

        
readingLevel()